Readability.js

```diff
@@ -961,6 +961,18 @@ Readability.prototype = {
           return null;
         }
       } else {
+        // Find out text direction from ancestors of final top candidate.
+        var ancestors = [parentOfTopCandidate].concat(this._getNodeAncestors(parentOfTopCandidate));
```

Comment:
Got it. I've updated patch and added tests for it. Please review the patch, thanks.